### ISSUE TITLE
Simplify metadata parsing slightly

### ISSUE THREAD

**Issue by tabatkins** [tabatkins/bikeshed#195]
I'm looking at some mild refactoring, and think I want to split out metadata parsing from the other pre blocks.  (It really needs to be the first bit of processing that occurs.)

Right now `<pre class=metadata>` has the same flexibility as the other specially-processed `<pre>` elements; you can put the first line of content on the same line as the opening `<pre>`, put the `</pre>` on the same line as the last line of content, indent the entire block, etc.

I'd like to make it a bit simpler, so you _must_ follow this pattern:

```
<pre class="metadata">
Foo: bar
Baz: Qux
</pre>
```

The `<pre>` tags must be on separate lines from the content, and it must be up against the edge.

This somewhat simplifies the parsing, but it also allows for a new feature that I've seen in something else with the same syntax: If you start a line with whitespace, it'll be considered a continuation of the previous line.  This simplifies multi-line Abstracts a bit, for example:

```
<pre class="metadata">
Abstract: This is the first line and
 then we wrap it onto the second line.

 And this is the second paragraph
 of the abstract.
Editor: Tab Atkins
</pre>
```

So basically, @mikewest and @dirkschulze, are you guys using indented or otherwise weird metadata blocks?  If so, please change before I make this edit. ^_^
